### PR TITLE
HDDS-5106. [FSO] Avoid using V1 postfixes for prefix related classes

### PR DESCRIPTION
**Pull request by rakeshadr**
## What changes were proposed in this pull request?

Rename V1 postfixes to appropriate feature name "`FSO`" - `FileSystemOptimized`. 

`PREFIX` metadata layout + `enableFSPaths "true"` = together will execute FSO feature code path.

Note: The changes are quite large due to the large amount of keyword renaming "V1" to "FSO" but its trivial.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5106

## How was this patch tested?

Existing test cases.

### CODE REVIEW

**Comment by rakeshadr** [comment]
**Note**: The changes are quite large due to the large amount of keyword renaming "V1" to "FSOB" but its trivial.

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmKeyInfoCodec.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  * Codec to encode OmKeyInfo as byte array.
  *
  * <p>
- * If the layout version "ozone.om.layout.version" is V1 and
+ * If the layout version "ozone.om.layout.version" is PREFIX and
```
Comment (ayushtkn) [inline hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmKeyInfoCodec.java:37]: Is this `ozone.om.metadata.layout` or `ozone.om.layout.version`

**Review by ayushtkn — APPROVED** [review]
There are couple of changes in variable names as well, like `omResponse` -> `omResp`, `omRequest` -> `omReq`, `isDirectory` -> `isDir`. I think they were intentional.

Changes LGTM.

**Comment by rakeshadr** [comment]
> There are couple of changes in variable names as well, like `omResponse` -> `omResp`, `omRequest` -> `omReq`, `isDirectory` -> `isDir`. I think they were intentional.
> 
> Changes LGTM.

Yes, I made those changes intentionally to reduce the statement length. I think, these var names are still readable :-)

Thanks @ayushtkn for the reviews.

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmKeyInfoCodec.java (hunk 1 above)
Comment (rakeshadr) [reply hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmKeyInfoCodec.java:37]: @ayushtkn Thanks for pointing out this. Its `ozone.om.metadata.layout` , I will update it.

**Comment by rakeshadr** [comment]
Thanks @ayushtkn, @mukul1987  for the reviews. I've merged it to the branch.

**Pull request merged**